packages/utilities/src/lib/sleep.ts

```diff
@@ -0,0 +1,65 @@
+export interface SleepOptions {
+	/**
+	 * When provided the corresponding `AbortController` can be used to cancel an asynchronous action.
+	 */
+	signal?: AbortSignal | undefined;
+
+	/**
+	 * Set to `false` to indicate that the scheduled `Timeout`
+	 * should not require the Node.js event loop to remain active.
+	 * @default true
+	 */
+	ref?: boolean | undefined;
+}
+
+export class AbortError extends Error {
+	public code: string;
+	public constructor(
+		message?: string,
+		options?: {
+			cause?: unknown;
+			code?: string;
+		}
+	) {
```

Comment:
Following the other review:
```suggestion
	public constructor(message?: string, options?: { cause?: unknown }) {
```